source/containerimage/pull.go

```diff
@@ -65,10 +77,17 @@ func NewSource(opt SourceOpt) (*Source, error) {
 }
 
 func (is *Source) ID() string {
+	if is.Source == SourceOCILayout {
+		return srctypes.OCIScheme
+	}
 	return srctypes.DockerImageScheme
 }
 
 func (is *Source) ResolveImageConfig(ctx context.Context, ref string, opt llb.ResolveImageConfigOpt, sm *session.Manager, g session.Group) (digest.Digest, []byte, error) {
+	if is.Source != SourceRegistry {
```

Comment:
I don't think you want this limitation. You will get unexpected results if `ImageConfig` is not resolved.

Reply:
I think we might need it for now. If I read this correctly, `ResolveImageConfig()` takes a `ref` (e.g. `docker.io/ibrary/nginx:latest`), and gets the bytes and hash of the config from the manifest.

By the way, how do you figure out which platform to use for a multi-arch index?

If I read it incorrectly - that it might also take a full hash, e.g. `docker.io/foo/bar@sha256:abcdef1233` and get the config, then, sure, we can add it.